tasks/sync-granule/index.js

```diff
@@ -95,6 +96,7 @@ function syncGranule(event) {
   const downloadBucket = config.downloadBucket;
   const syncChecksumFiles = config.syncChecksumFiles;
   const duplicateHandling = duplicateHandlingType(event);
+  const workflowStartTime = config.workflowStartTime > now ? now : config.workflowStartTime;
```

Comment:
This logic is a bit confusing, even if it does work. `config.workflowStartTime` could be undefined, so `undefined > Date.now()` is a bit awkward even if it does yield a result.

How about this:

```
const workflowStartTime  = config.workflowStartTime ? Math.min(config.workflowStartTime,now) : now
```

We only try to use `config.workflowStartTime` if it has a value. If it does, we only use it if less than or equal to `now`, and if there is no value we just fall back to `now`

Reply:
how could workflow_start_time be a future time?

Reply:
People can set `config.workflowStartTime` to whatever they want in the workflow config, including a hard-coded future timestamp. It's definitely an edge case, but relatively simple to defend against

Reply:
good point. updated.